std/algorithm/iteration.d

```diff
@@ -2706,21 +2706,24 @@ if (isInputRange!RoR && isInputRange!(ElementType!RoR))
 
         private enum popFrontEmptyElements = q{
             // Skip over empty subranges.
-            if (_items.empty) goto end;
-            while (_items.front.empty)
+            while (!_items.empty && _items.front.empty)
             {
                 _items.popFront();
-                if (_items.empty) goto end;
             }
-            // We cannot export .save method unless we ensure subranges are not
-            // consumed when a .save'd copy of ourselves is iterated over. So
-            // we need to .save each subrange we traverse.
-            static if (isForwardRange!RoR && isForwardRange!(ElementType!RoR))
-                _current = _items.front.save;
+            if (!_items.empty)
+            {
+                // We cannot export .save method unless we ensure subranges are not
+                // consumed when a .save'd copy of ourselves is iterated over. So
+                // we need to .save each subrange we traverse.
+                static if (isForwardRange!RoR && isForwardRange!(ElementType!RoR))
+                    _current = _items.front.save;
+                else
+                    _current = _items.front;
+            }
             else
-                _current = _items.front;
-        end:
-            assert(1); // required to avoid 'EOF instead of statement' error
+            {
+                _current = typeof(_current).init;
```

Comment:
Why is this required?

Reply:
The compiler requires that `_current` is initialised on all possible paths.